linc_convert/utils/spool.py

```diff
@@ -0,0 +1,185 @@
+
+        except (KeyError, IndexError, ValueError) as e:
+            warnings.warn(
+                f"ERROR: Unable to extract frame information from info file. {e}")
+            return None
+        num_frames_per_spool = int(self.config['multiimage']['ImagesPerFile'])
+        num_frames_to_load = num_total_frames - num_bg_frames
+        num_frames_to_load = ceildiv(num_frames_to_load, num_frames_per_spool)
+        self.spool_files = self.spool_files[:num_frames_to_load]
+
+    def _make_filename_from_spool_set(self, spool_entry):
+        return os.path.join(self.parent, spool_entry)
+
+    @property
+    def entries(self):
+        return self.spool_set
+
+    def _what_spool_format(self):
+        if 'Spooled files.sifx' in self.entries:
+            format = 'zyla'
+        else:
+            raise TypeError("Unknown or unsupported spool file format")
+
+        self.format = format
+
+    def _list_spool_files(self):
+        return sorted(
+            tuple(
+                [x for x in self.entries if '0spool.dat' in x]
+            )
+        )
+
+    def _get_acquisitionparameters_str(self):
+        file = self._make_filename_from_spool_set('acquisitionmetadata.ini')
+        with open(file, 'rb') as f:
+            ini = f.read()
+        ini = ini.decode('UTF-8-sig')  # Encoding for acquisitionmetadata.ini
+        self.acquisitionparameters_str = ini
+
+    def _get_config(self):
+        buf = io.StringIO(self.acquisitionparameters_str)
+        self.config = configparser.ConfigParser()
+        self.config.read_file(buf)
+
+    def _extract_config_values(self):
+        self.acquisition_metadata = {'height': self.config.getint('data', 'AOIHeight'),
+                                     'width': self.config.getint('data', 'AOIWidth'),
+                                     'stride': self.config.getint('data', 'AOIStride')}
+        # ini info
+        dtype = self.config.get('data', 'PixelEncoding')
+
+        if dtype == 'Mono16':
+            dtype = np.dtype('uint16')
+        elif dtype == 'Mono8':
+            dtype = np.dtype('uint8')
+
+        self.acquisition_metadata['dtype'] = dtype
+        self.dtype = dtype
+
+        self.spool_nbytes = self.config.getint('data', 'ImageSizeBytes')
+        self.acquisition_metadata['nbytes'] = self.spool_nbytes
+
+        self.acquisition_metadata['images'] = self.config.getint('multiimage',
+                                                                 'ImagesPerFile')
+
+        numDepths = self.acquisition_metadata['height']
+        numLatPix = self.acquisition_metadata['stride'] // 2
+        imageBytes = self.acquisition_metadata['nbytes']
+        numFramesPerSpool = self.acquisition_metadata['images']
+
+        numRows = numDepths + 2
+
+        if numDepths % 2:  # if there is an odd number of rows ->  KPEDIT - odd rows means 1 less column for some reason
+            numRows = numDepths + 1
+
```

Comment:
Functionality is the same.  But just changing for readability.

```suggestion
        if numDepths % 2:  # if there is an odd number of rows ->  KPEDIT - odd rows means 1 less column for some reason
            numRows = numDepths + 1
        else:
            numRows = numDepths + 2

```